core/src/main/java/feign/ReflectiveAsyncFeign.java

```diff
@@ -0,0 +1,126 @@
+
+package feign;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Proxy;
+import java.lang.reflect.Type;
+import java.lang.reflect.WildcardType;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+public class ReflectiveAsyncFeign<C> extends AsyncFeign<C> {
+
+	private static class MethodInfo {
```

Comment:
`MethodMetadata` already exists to serve this purpose.  We should consider refactoring that component before creating additional components.

Reply:
Agreed - it was doing much too much for my needs, hence the duplication. Much preferable to refactor.